### PR TITLE
CDpypi

### PR DESCRIPTION
**Pull request by MaxOstrowski**
Add continuous deployment on [test.]pypi.org

### CODE REVIEW

File: .github/workflows/buildwheel.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    name: Build and upload
+    runs-on: ubuntu-latest
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
```
Comment (MaxOstrowski) [inline .github/workflows/buildwheel.yml:21]: I'm unsure if we need to specify a specific version/more than one version here?

Comment (MaxOstrowski) [reply .github/workflows/buildwheel.yml:21]: This semms to be fine. Can be merged I think.

**Comment by MaxOstrowski** [comment]
Is there any disadvantage of having reusable workflows? I thought it might be easier to extend and maintain (e.g. adding conda, etc...) But I don't have a strong opinion about it. This is the first time I use reusables.

The version check sounds like a useful thing to have. I don't think that the behvaiour for test.pypi.org should be different, no ? You can't overwrite versions there too, no ?

**Comment by rkaminsk** [comment]
The deployment section still needs cleanup.

**Comment by rkaminsk** [comment]
I was exploring a bit how `setuptools_scm` can help us to ease the deployment process.

The current state is here and I am actually quite satisfied with it. Please have a link at the links below:

- https://github.com/rkaminsk/fuzzy-waddle
- https://test.pypi.org/project/fuzzy-waddle/

To summarize, I did the following:
- merged `setup.cfg` into `pyproject.toml` because the latter seems to be the thing
- the deployment process is in `deploy.yml` which
  - automatically deploys to pypi.org (url still has to be changed) when pushing a version tag
    - the version will automatically be extracted from the tag
  - can be used to manually deploy to test.pypi
    - the version will automatically be extracted from the last version tag
    - a `.devN` suffix will be added where `N` is the number of commits since the last version tag

Max's proposal was using a reusable workflow to run the tests before the deployment. This can easily be added back. Just let me know what you prefer.

**Comment by rkaminsk** [comment]
@MaxOstrowski: let's continue from #31

**Pull request closed without merging**